z3c/dependencychecker/dotted_name.py

```diff
@@ -25,8 +25,12 @@ def from_requirement(cls, requirement, file_path=None):
         """A requirement in this method's context is a
         pkg_resources.Requirement
         """
+        # some pypi packages have a python- prefix that is not part of the
```

Comment:
Well, this is just one of the many conventions. A django prefix is also quite often used. But sometimes it really ought to keep the django prefix and sometimes it needs to be used without.

Likewise, stripping the python prefix seems very dangerous to me.

Reply:
I'm getting the impression that the new implementation treats imports and requirements the same. Both are converted to a single dotted name. So there's no room for:

- Trying to import something to figure out which requirement name would match.
- A package providing multiple top-level paths (like some zope "meta-packages" do).

So I think something a bit more explicit is needed here. A distinction between imports (which are really dotted names) and packages/requirements, which might have to be handled a little bit special.

Reply:
In my tests, I've seen a package "pyjwt" that is imported as 'jwt'.

Package `fake-factory` as `import faker`.

django-appconf => appconf

django-oidc-provider => oidc_provider

django-tls => tls

So if you'd also handle 'py' and 'django' in the same way, I'd see less unrecognized packages. But I'm quite afraid of ruining some real python-something or django-something packages.